server/sonar-db-dao/src/main/java/org/sonar/db/purge/PurgeDao.java

```diff
@@ -67,6 +68,22 @@ public void purge(DbSession session, PurgeConfiguration conf, PurgeListener list
     purgeAnalyses(commands, rootUuid);
     purgeDisabledComponents(session, conf, listener);
     deleteOldClosedIssues(conf, mapper, listener);
+    purgeStaleBranches(commands, conf, mapper, rootUuid);
+  }
+
+  private static void purgeStaleBranches(PurgeCommands commands, PurgeConfiguration conf, PurgeMapper mapper, String rootUuid) {
+    Optional<Date> maxDate = conf.maxLiveDateOfInactiveShortLivingBranches();
+    if (!maxDate.isPresent()) {
+      // not available if branch plugin is not installed
```

Comment:
The purge should be done even without plugin => if the plugin is removed, we should purge data to not keep branches forever